services/registration/service/mail_service.go

```diff
@@ -25,3 +27,35 @@ func SendUserMail(id string, template string) error {
 
 	return err
 }
+
+/*
+	Add user with given id, to the specified mailing list.
+	If the mailing list does not exist, it creates a new list with the user.
+*/
+func AddUserToMailList(user_id string, mail_list_id string) error {
+	add_to_mail_list_url := fmt.Sprintf("%s/mail/list/add/", config.MAIL_SERVICE)
+
+	mail_list := models.MailList{
+		ID:      mail_list_id,
+		UserIDs: []string{user_id},
+	}
+
+	update_request_body := bytes.Buffer{}
+	json.NewEncoder(&update_request_body).Encode(&mail_list)
+
+	status, err := apirequest.Post(add_to_mail_list_url, &update_request_body, nil)
+
+	if err != nil || status != http.StatusOK {
```

Comment:
If `err != nil` that indicates a different error than `status != http.StatusOK` and need to be handed differently.

You can add
```
if err != nil {
    return err
}
```

above that check for status.